temporal/server_options.go

```diff
@@ -91,12 +92,25 @@ func (so *serverOptions) loadAndValidate() error {
 }
 
 func (so *serverOptions) loadConfig() error {
```

Comment:
Looks like we need the same check that we have in `cmd/server/main.go` to ensure that `env`, `configDir`, etc aren't set along with `configFilePath`.